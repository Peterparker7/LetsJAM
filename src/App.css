* {
  box-sizing: border-box;
  font-family: "Noto Sans TC", sans-serif;
}

* {
  margin: 0;
  padding: 0;
  border: 0;
  outline: 0;
  font-size: 100%;
  vertical-align: baseline;
  background: transparent;
  /* outline: 1px solid black; */
}

a {
  text-decoration: none;
  color: black;
}

/* html {
  overflow: hidden;
  height: 100%;
}
body {
  overflow: auto;
  height: 100%;
} */
body {
  overscroll-behavior-y: none;
}

.rmsc {
  --rmsc-main: #4285f4;
  --rmsc-hover: #f1f3f5;
  --rmsc-selected: #e2e6ea;
  --rmsc-border: #ccc;
  --rmsc-gray: #aaa;
  /* --rmsc-bg: #fff; */
  --rmsc-p: 10px; /* Spacing */
  --rmsc-radius: 4px; /* Radius */
  --rmsc-h: 40px !important; /* Height */
  width: 100%;
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
