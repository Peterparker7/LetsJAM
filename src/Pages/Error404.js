import styled from "styled-components";
import React from "react";
// import error404 from "../images/404-error.svg";
import { useHistory } from "react-router-dom";
import { keyframes } from "styled-components";

const NeonShine = keyframes`
  0% {opacity: 1}
  10%{opacity:0.7;}
  20%{opacity:1;}
  60%{opacity:1;}
  70%{opacity:0.7;}
  85%{opacity:1;}
  100% {opacity: 1}
`;
const NeonShine2 = keyframes`
  0% {opacity: 1}
1%{opacity:0}
2%{opacity:1}
3%{opacity:0}
4%{opacity:1}
`;
const Container = styled.div`
  width: 100%;
  background: #121212;
  height: 100vh;
  padding-top: 100px;
`;
// const ErrorIconContainer = styled.div`
//   width: 250px;
//   height: 250px;
//   margin: 0px auto;
//   align-items: center;
// `;

// const ErrorIcon = styled.img`
//   width: 100%;
// `;
// const ErrorText = styled.div`
//   color: white;
//   font-size: 24px;
//   font-weight: 700;
//   text-align: center;
// `;
const ButtonField = styled.div`
  margin: 20px auto;
  width: 200px;
  text-align: center;
`;
const MainPageBtn = styled.button`
  margin-top: 20px;
  padding: 12px 40px;
  background: #43e8d8;
  border-radius: 8px;
  color: black;
  text-align: center;
  font-size: 16px;
  font-weight: 600;
  background: #43e8d8;
  cursor: pointer;
  transition: 0.2s;
  &:hover {
    background: #4cffee;
    transform: translateY(-2px);
    box-shadow: 0 0 20px #43e8d8;
  }
`;
const NeonBox = styled.div`
  margin: 0 auto;
  border: 4px solid white;
  width: 350px;
  padding: 20px;
  box-shadow: 0 0 10px #43e8d8, 0 0 20px #43e8d8, inset 0 0 10px #43e8d8,
    inset 0 0 20px #43e8d8;
  animation: ${NeonShine2} 10s 1s linear infinite;
`;
const NeonText = styled.div`
  text-align: center;
  color: white;
  font-size: 36px;
  font-weight: 600;
  text-shadow: 0 0 5px rgba(255, 65, 65, 1), 0 0 10px rgba(255, 65, 65, 1),
    0 0 20px rgba(255, 65, 65, 1), 0 0 40px rgba(255, 65, 65, 1),
    0 0 60px rgba(255, 65, 65, 1);
  animation: ${NeonShine} 10s 1s linear infinite;
`;

export default function Error404() {
  const history = useHistory();

  function backToHome() {
    history.push("/");
  }
  return (
    <Container>
      {/* <ErrorIconContainer>
        <ErrorIcon src={error404} alt="" />
      </ErrorIconContainer>
      <ErrorText>Oops~ 找不到此頁</ErrorText> */}
      <NeonBox>
        <NeonText>404</NeonText>
        <NeonText>Page Not Found</NeonText>
      </NeonBox>
      <ButtonField>
        <MainPageBtn onClick={backToHome} data-testid="homeButton">
          回Let's JAM
        </MainPageBtn>
      </ButtonField>
    </Container>
  );
}
