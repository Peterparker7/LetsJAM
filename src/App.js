import "./App.css";
import Main from "./Main";
import Create from "./Create";
import Detail from "./Detail";
import Profile from "./Profile";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";

function App() {
  return (
    <div className="App">
      <Router>
        {/* <Header/> */}
        <Switch>
          <Route exact path="/activities">
            <Main />
          </Route>
          <Route exact path="/activities/create">
            <Create />
          </Route>
          <Route exact path="/activities/profile">
            <Profile />
          </Route>
          <Route exact path="/activities/:id">
            <Detail />
          </Route>
        </Switch>
      </Router>
    </div>
  );
}

export default App;
