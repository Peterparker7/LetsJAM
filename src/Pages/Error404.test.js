import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";

import renderer from "react-test-renderer";
import App from "../App";
import Error404 from "./Error404";

describe("Error404", () => {
  render(<Error404></Error404>);
  it("should render home button", async () => {
    // const mockFn = jest.fn();
    const buttonElement = screen.getByTestId("homeButton");
    // buttonElement.onclick = { mockFn };
    expect(buttonElement).toBeInTheDocument();

    // userEvent.click(buttonElement);
    // expect(mockFn).toBeCalledTimes(1);
  });
  it("should render correctly", () => {
    const tree = renderer.create(<Error404 />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
