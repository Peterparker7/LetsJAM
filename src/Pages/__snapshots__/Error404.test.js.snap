// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 should render correctly 1`] = `
<div
  className="sc-kQfVtO bVrxVR"
>
  <div
    className="sc-bfmRie haygEl"
  >
    <div
      className="sc-eEihID fpbWQc"
    >
      404
    </div>
    <div
      className="sc-eEihID fpbWQc"
    >
      Page Not Found
    </div>
  </div>
  <div
    className="sc-ektJcd iffmnv"
  >
    <button
      className="sc-khIimk jAqwDB"
      data-testid="homeButton"
      onClick={[Function]}
    >
      å›žLet's JAM
    </button>
  </div>
</div>
`;
